plugins/scaffolder-backend-module-gitlab/src/actions/gitlabIssueEdit.ts

```diff
@@ -0,0 +1,245 @@
+    })
+    .optional(),
+  dueDate: z
+    .string()
+    .describe(
+      'The due date. Date time string in the format YYYY-MM-DD, for example 2016-03-11.',
+    )
+    .regex(/^\d{4}-\d{2}-\d{2}$/, 'Invalid date format. Use YYYY-MM-DD')
+    .optional(),
+  epicId: z
+    .number({
+      description:
+        'ID of the epic to add the issue to. Valid values are greater than or equal to 0.',
+    })
+    .min(0, 'Valid values should be equal or greater than zero')
+    .optional(),
+  issueType: z
+    .nativeEnum(IssueType, {
+      description:
+        'Updates the type of issue. One of issue, incident, test_case or task.',
+    })
+    .optional(),
+  labels: z
+    .string({
+      description:
+        'Comma-separated label names for an issue. Set to an empty string to unassign all labels. If a label does not already exist, this creates a new project label and assigns it to the issue.',
+    })
+    .optional(),
+  milestoneId: z
+    .number({
+      description:
+        'The global ID of a milestone to assign the issue to. Set to 0 or provide an empty value to unassign a milestone',
+    })
+    .optional(),
+  removeLabels: z
+    .string({
+      description: 'Comma-separated label names to remove from an issue.',
+    })
+    .optional(),
+  stateEvent: z
+    .nativeEnum(IssueStateEvent, {
+      description:
+        'The state event of an issue. To close the issue, use close, and to reopen it, use reopen.',
+    })
+    .optional(),
+  title: z.string().describe('The title of an issue.').optional(),
+  updatedAt: z
+    .string()
+    .describe(
+      'When the issue was updated. Date time string, ISO 8601 formatted',
+    )
+    .regex(
+      /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(?:\.\d{3})?Z$/,
+      'Invalid date format. Use YYYY-MM-DDTHH:mm:ssZ or YYYY-MM-DDTHH:mm:ss.SSSZ',
+    )
+    .optional(),
+  weight: z
+    .number({ description: 'The issue weight' })
+    .min(0, 'Valid values should be equal or greater than zero')
+    .max(10, 'Valid values should be equal or less than 10')
+    .optional(),
+});
+
+const editIssueOutputProperties = z.object({
+  issueUrl: z.string({ description: 'Issue WebUrl' }),
+  projectId: z.number({
+    description: 'The project id the issue belongs to WebUrl',
+  }),
+  issueId: z.number({ description: 'The issues Id' }),
+  issueIid: z.number({
+    description: "The issues internal ID of a project's issue",
+  }),
+  state: z.string({ description: 'The state event of an issue' }),
+  title: z.string({ description: 'The title of an issue.' }),
+  updatedAt: z.string({ description: 'The last updated time of the issue.' }),
+});
+
+/**
+ * Creates a `gitlab:issues:edit` Scaffolder action.
+ *
+ * @param options - Templating configuration.
+ * @public
+ */
+export const editGitlabIssueAction = (options: {
+  integrations: ScmIntegrationRegistry;
+}) => {
+  const { integrations } = options;
+  return createTemplateAction({
+    id: 'gitlab:issues:edit',
```

Comment:
For consistency with other actions, it would be better to keep the name as non-plural: `gitlab:issue:edit`

Reply:
Agree - I can't change them from work so will have to wait until tonight.

Reply:
Action updated to remove plural to remain consistent.